src/main/java/software/amazon/cloudformation/resource/ResourceTypeSchema.java

```diff
@@ -145,4 +148,48 @@ public static ResourceTypeSchema load(final JSONObject schemaJson) {
     public Map<String, Object> getUnprocessedProperties() {
         return Collections.unmodifiableMap(this.unprocessedProperties);
     }
+
+    public boolean hasWriteOnlyProperties(final JSONObject resourceModel) {
+        return this.getWriteOnlyPropertiesAsStrings().stream()
+            .anyMatch(writeOnlyProperty -> hasProperty(RegExUtils.removeFirst(writeOnlyProperty, "/properties"), resourceModel));
+    }
+
+    public boolean hasProperty(final String property, final JSONObject resourceModel) {
+        return hasProperty(new JSONPointer(property), resourceModel);
+    }
+
+    public boolean hasProperty(final JSONPointer property, final JSONObject resourceModel) {
+        try {
+            return property.queryFrom(resourceModel) != null;
+        } catch (JSONPointerException e) {
+            return false;
+        }
+    }
+
+    public void removeWriteOnlyProperties(final JSONObject resourceModel) {
+        this.getWriteOnlyPropertiesAsStrings().stream()
+            .forEach(writeOnlyProp -> removeProperty(new JSONPointer(RegExUtils.removeFirst(writeOnlyProp, "/properties")),
```

Comment:
I think you can probably write something to remove properties without pulling in `RegExUtils`/`commons-lang3`. I'm all for DRY, but this is a bit silly and will further bloat the classpath while the Java plugin still depends on this library.

Reply:
Will do good idea